plugins/scaffolder-backend/src/util/checkPermissions.ts

```diff
@@ -21,19 +21,37 @@ import { NotAllowedError } from '@backstage/errors';
 import {
   AuthorizeResult,
   BasicPermission,
+  PolicyDecision,
+  ResourcePermission,
 } from '@backstage/plugin-permission-common';
 
-export type checkPermissionOptions = {
+export type checkBasicPermissionOptions = {
   credentials: BackstageCredentials;
   permissions: BasicPermission[];
   permissionService?: PermissionsService;
+  throwError?: boolean;
+};
+
+export type checkTaskPermissionOptions = {
+  credentials: BackstageCredentials;
+  permission: ResourcePermission;
+  permissionService?: PermissionsService;
+  createdBy?: string;
+  isTaskAuthorized: (
+    decision: PolicyDecision,
+    resource: string | undefined,
+  ) => boolean;
 };
 
 /**
- * Does a basic check on permissions. Throws 403 error if any permission responds with AuthorizeResult.DENY
+ * Does a basic check on permissions.
+ * If throwError is set to true, throws 403 error if any permission responds with AuthorizeResult.DENY
+ * Otherwise, it will return false instead of throwing an error
```

Comment:
is there any case where `throwError: false` is passed? I couldn't find it

Reply:
Ah that might have been some leftover code from when I had an explicit admin permission. Let me revert this back to what it was before.